src/ui/Triggers/Triggers.ts

```diff
@@ -78,109 +96,106 @@ export class Triggers extends Component {
     Assert.exists(data.results);
 
     $$(this.element).empty();
-    this.notifications.length = 0;
-    let showElement = false;
+    this.notifications = [];
 
-    if (data.results.triggers === undefined) {
-      $$(this.element).toggleClass('coveo-visible', showElement);
+    if (Utils.isNullOrUndefined(data.results.triggers)) {
+      $$(this.element).removeClass('coveo-visible');
       return;
     }
 
-    this.executeTriggers(data.results.triggers, 'notify', (trigger: ITriggerNotify) => {
-      this.usageAnalytics.logCustomEvent<IAnalyticsTriggerNotify>(
-        analyticsActionCauseList.triggerNotify,
-        {
-          notification: trigger.content
-        },
-        this.element
-      );
-
-      this.notifications.push(trigger.content);
-      this.element.appendChild($$('div', { className: 'coveo-trigger-notify' }, trigger.content).el);
-
-      showElement = true;
+    const notifyTriggers: ITriggerNotify[] = [];
+    let redirectTrigger: ITriggerRedirect | null = null;
+    let queryTrigger: ITriggerQuery | null = null;
+    let executeTrigger: ITriggerExecute | null = null;
+    data.results.triggers.forEach(trigger => {
+      if (isTriggerNotify(trigger)) {
+        notifyTriggers.push(trigger);
+      } else if (isTriggerRedirect(trigger)) {
+        redirectTrigger = redirectTrigger || trigger;
+      } else if (isTriggerQuery(trigger)) {
+        queryTrigger = queryTrigger || trigger;
+      } else if (isTriggerExecute(trigger)) {
+        executeTrigger = executeTrigger || trigger;
+      }
     });
 
-    this.executeTriggers(
-      data.results.triggers,
-      'redirect',
-      (trigger: ITriggerRedirect) => {
-        this.usageAnalytics.logCustomEvent<IAnalyticsTriggerRedirect>(
-          analyticsActionCauseList.triggerRedirect,
-          {
-            redirectedTo: trigger.content
-          },
-          this.element
-        );
+    if (notifyTriggers.length) {
+      this.processNotifyTriggers(notifyTriggers);
+      $$(this.element).addClass('coveo-visible');
+    }
+    redirectTrigger && this.processRedirectTrigger(redirectTrigger);
+    queryTrigger && this.processQueryTrigger(queryTrigger);
+    executeTrigger && this.processExecuteTrigger(executeTrigger, data.query);
+  }
 
-        this._window.location.replace(trigger.content);
+  private processNotifyTriggers(triggers: ITriggerNotify[]) {
+    this.usageAnalytics.logCustomEvent<IAnalyticsTriggerNotify>(
```

Comment:
The only trigger logic that changed is the one for the notify triggers. The rest are copy and pasted.